tests/main/views/test_services.py

```diff
@@ -1674,3 +1674,132 @@ def test_get_service_returns_correct_unavailability_audit_if_disabled_but_framew
+                )),
+            )
+
+    def test_non_existent_service(self, search_api_client):
+        response = self.client.post(
+            '/services/9876987698760000/revert',
+            content_type='application/json',
+            data=json.dumps({"updated_by": "papli@example.com", "archivedServiceId": self.archived_service_ids[0]}),
+        )
+        assert response.status_code == 404
+
+        with self.app.app_context():
+            assert not AuditEvent.query.filter(AuditEvent.type == AuditTypes.update_service.name).all()
+            assert ArchivedService.query.count() == 4
+            assert search_api_client.index.called is False
+
+    def test_archived_service_mismatch(self, search_api_client):
+        response = self.client.post(
+            '/services/1234123412340001/revert',
+            content_type='application/json',
+            data=json.dumps({"updated_by": "papli@example.com", "archivedServiceId": self.archived_service_ids[3]}),
+        )
+        assert response.status_code == 400
+        assert "correspond" in json.loads(response.get_data())['error']
+
+        with self.app.app_context():
+            assert not AuditEvent.query.filter(AuditEvent.type == AuditTypes.update_service.name).all()
+            assert Service.query.filter(
+                Service.service_id == "1234123412340001"
+            ).one().data["serviceSummary"] == "Probably the best cloud service in the world"
+            assert ArchivedService.query.count() == 4
+            assert search_api_client.index.called is False
+
+    def test_non_existent_archived_service(self, search_api_client):
+        response = self.client.post(
+            '/services/1234123412340001/revert',
+            content_type='application/json',
+            data=json.dumps({"updated_by": "papli@example.com", "archivedServiceId": 321321}),
+        )
+        assert response.status_code == 400
+        assert "ArchivedService" in json.loads(response.get_data())['error']
+
+        with self.app.app_context():
+            assert not AuditEvent.query.filter(AuditEvent.type == AuditTypes.update_service.name).all()
+            assert Service.query.filter(
+                Service.service_id == "1234123412340001"
+            ).one().data["serviceSummary"] == "Probably the best cloud service in the world"
+            assert ArchivedService.query.count() == 4
+            assert search_api_client.index.called is False
+
+    def test_archived_service_doesnt_validate(self, search_api_client):
+        response = self.client.post(
+            '/services/1234123412340001/revert',
+            content_type='application/json',
+            data=json.dumps({"updated_by": "papli@example.com", "archivedServiceId": self.archived_service_ids[2]}),
+        )
+        assert response.status_code == 400
+        assert json.loads(response.get_data())['error'] == {'serviceSummary': 'answer_required'}
+
+        with self.app.app_context():
```

Comment:
This set of assertions is repeated 3 times. Could you pull it out to a shared thing?

Reply:
maybe, but I don't know if it would improve the readability